t/t1028-mf-priority-projects.t

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+
+test_description='test validating and setting project names in priority plugin'
+
+. `dirname $0`/sharness.sh
+MULTI_FACTOR_PRIORITY=${FLUX_BUILD_DIR}/src/plugins/.libs/mf_priority.so
+SUBMIT_AS=${SHARNESS_TEST_SRCDIR}/scripts/submit_as.py
+DB_PATH=$(pwd)/FluxAccountingTest.db
+
+export TEST_UNDER_FLUX_NO_JOB_EXEC=y
+export TEST_UNDER_FLUX_SCHED_SIMPLE_MODE="limited=1"
+test_under_flux 1 job
+
+flux setattr log-stderr-level 1
+
+test_expect_success 'load multi-factor priority plugin' '
+	flux jobtap load -r .priority-default ${MULTI_FACTOR_PRIORITY}
+'
+
+test_expect_success 'check that mf_priority plugin is loaded' '
+	flux jobtap list | grep mf_priority
+'
+
+test_expect_success 'create flux-accounting DB' '
+	flux account -p $(pwd)/FluxAccountingTest.db create-db
+'
+
+test_expect_success 'start flux-accounting service' '
+	flux account-service -p ${DB_PATH} -t
+'
+
+test_expect_success 'add banks to the DB' '
+	flux account -p ${DB_PATH} add-bank root 1 &&
+	flux account -p ${DB_PATH} add-bank --parent-bank=root account1 1
+'
+
+test_expect_success 'add projects to the DB' '
+	flux account -p ${DB_PATH} add-project projectA &&
+	flux account -p ${DB_PATH} add-project projectB &&
+	flux account -p ${DB_PATH} add-project projectC
+'
+
+test_expect_success 'submit a job under a user before plugin gets updated' '
+	job0=$(flux python ${SUBMIT_AS} 1003 hostname) &&
+	test $(flux jobs -no {state} ${job0}) = PRIORITY
+'
+
+test_expect_success 'add user to flux-accounting DB and to plugin; job transitions to RUN' '
+	flux account -p ${DB_PATH} add-user --username=user1003 --userid=1003 --bank=account1 &&
+	flux account-priority-update -p ${DB_PATH} &&
+	test $(flux jobs -no {state} ${job0}) = RUN
+'
+
+test_expect_success 'check that project gets updated for submitted job' '
+	flux job info $job0 eventlog > eventlog.out &&
+	grep "{\"attributes.system.project\":\"\*\"}" eventlog.out &&
+	flux job cancel $job0
+'
+
+test_expect_success 'add a user with a list of projects to the DB' '
+	flux account -p ${DB_PATH} add-user --username=user1001 --userid=1001 --bank=account1 --projects="projectA,projectB"
+'
+
+test_expect_success 'send flux-accounting DB information to the plugin' '
+	flux account-priority-update -p $(pwd)/FluxAccountingTest.db
+'
+
+test_expect_success 'successfully submit a job under a valid project' '
+	job1=$(flux python ${SUBMIT_AS} 1001 --setattr=system.project=projectA hostname) &&
+	flux job wait-event -f json $job1 priority | jq '.context.priority' > job1.test &&
+	cat <<-EOF >job1.expected &&
+	50000
+	EOF
+	test_cmp job1.expected job1.test &&
+	flux job info $job1 jobspec > jobspec.out &&
+	grep "projectA" jobspec.out &&
+	flux job cancel $job1
+'
+
+test_expect_success 'submit a job under a project that does not exist' '
+	test_must_fail flux python ${SUBMIT_AS} 1001 --setattr=system.project=projectFOO \
+		hostname > project_dne.out 2>&1 &&
+	test_debug "project_dne.out" &&
```

Comment:
Just noticed this while trying to run this test with `-d -v` -- `test_debug()` takes a command to run, not a file. Since there's no command `project_dne.out`, this line causes a failure. Change to 
```
test_debug "cat project_dne.out" &&
```
here and in the other use of `test_debug` in this sharness test. (Also perhaps audit the rest of the testsuite)